src/System.Runtime.Environment/src/OSName.cs

```diff
@@ -0,0 +1,39 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+namespace System.Runtime.InteropServices
+{
+    public struct OSName : IEquatable<OSName>
+    {
+        private string _name;
+
+        private const string WindowsName = "windows";
+        private const string LinuxName = "linux";
+        private const string OSXName = "osx";
+
+        private static readonly OSName s_windows = new OSName(WindowsName);
+        private static readonly OSName s_linux = new OSName(LinuxName);
+        private static readonly OSName s_osx = new OSName(OSXName);
+
+        public OSName(string name)
+        {
+            if (name.Equals("unix", StringComparison.OrdinalIgnoreCase))
```

Comment:
I don't see a reason for allowing comparing equal if the user passes a different form, i.e. if they pass different form, the comparison will return false, and it's OK, IMHO.